Make tests pass

diveshuttam opened the pull request:
Helps with #63 

Though not all tests pass; specifically the ones mentioned in EasyGnuPG/egpg#56 don't pass.
Tests for the features added up to f181139480eefdac5fbc4a0e1797d9473993f1f0 pass.
Not tested/updated for #60 and #61 yet.

Also there are cases when the test don't pass for the first time (t03-init) I spring up the container and I get the error that key not found. But they pass the second time and further. What I have currently figured out is that there are certain files(keys) added to tests/gnupg after t03 which make them run the second and further time. (I'll work this out in a while)
This was the reason they didn't work the first time yesterday but when I copied the egpg/tests directory to pgpg and run a couple of times and they started working.

diveshuttam committed:
Make tests pass

this updates tests/code to make tests for verify, sign pass with gpgme too

dashohoxha commented:
> Also there are cases when the test don't pass for the first time (t03-init) I spring up the container and I get the error that key not found. But they pass the second time and further.

I think that I fixed it a month ago: https://github.com/EasyGnuPG/egpg/pull/43/files

diveshuttam commented:
>I think that I fixed it a month ago: https://github.com/EasyGnuPG/egpg/pull/43/files

Yes I have that commit in my tests and it is still creating problem. The first time I run test3, the keys are generated/renewed but the test fails; It passes subsequent times as the keys are now there in the tests/gnupg folder. You might not face this problem if you have ran these tests even once. But if you checkout all the files in test folder to the HEAD, they won't pass the first time. They will pass subsequent times. I think we need to have some key files added to the commit which are not already there.

dashohoxha commented:
You are right, we should find out what makes it fail the first time, and fix it.

diveshuttam committed:
update tests from egpg

EasyGnuPG/egpg#bc05ca527bc2c1d0573a321eea439ea194f7e95b

diveshuttam commented:
@dashohoxha as mentioned in another comment on #63 the problem was that the setup.sh was first copying keys (in tests/gnupg) to trash dir and then we were renewing keys in tests/gnupg only(inside t03 file) but not copying them to trash dir. I have moved the renew functionality to setup.sh only. This will help us in running tests in any order and also solves this problem. Though it breaks the existing code structure as I have added the statement to the last of setup.sh. Please suggest your thoughts about the code.

dashohoxha commented:
I have done something similar here: https://github.com/EasyGnuPG/egpg/pull/57/files
Please do it the same way, in order to avoid unnecessary divergences between egpg and pgpg.

diveshuttam commented:
@dashohoxha done. Please review.
